common/network-common/src/main/java/org/apache/spark/network/sasl/SaslRpcHandler.java

```diff
@@ -155,4 +180,48 @@ public void exceptionCaught(Throwable cause, TransportClient client) {
     delegate.exceptionCaught(cause, client);
   }
 
+  /**
+   * Negotiates AES based on complete {@link SparkSaslServer}. The keys need to be encrypted by
+   * sasl server.
+   */
+  private void negotiateAes(ByteBuffer message, RpcResponseCallback callback) {
+    // receive initial option from client
+    CipherOption cipherOption = CipherOption.decode(Unpooled.wrappedBuffer(message));
+    CipherTransformation transformation = CipherTransformation.fromName(cipherOption.cipherSuite);
+    Properties properties = new Properties();
```

Comment:
Do we need specific set the value for ConfigurationKeys.CHIMERA_CRYPTO_SECURE_RANDOM_CLASSES_KEY?
I think the default value should work.